pkg/scm/gitlab/context.go

```diff
@@ -158,3 +164,47 @@ func (c *Context) HasExecutedActionGroup(group string) bool {
 
 	return ok
 }
+
+// GetCodeOwners returns the eligible code owners for the merge request
+//
+// This is based on the elibible approvers in the rules of the merge request approval
+// state api.
+func (c *Context) GetCodeOwners() []scm.Actor {
+	owners := []scm.Actor{}
+
+	for _, rule := range c.MergeRequest.ApprovalState.Rules {
```

Comment:
Any `nil` guarding needed for MRs or projects without any approval rules/requirements?

Reply:
we won't have to, there will always be at least 1 rule of

```
            {
              "name": "All Members",
              "type": "ANY_APPROVER",
              "approvalsRequired": 0,
              "approved": true,
              "eligibleApprovers": []
            }
```